packages/core/src/indexing-store/realtimeStore.ts

```diff
@@ -81,49 +72,40 @@ export const getRealtimeIndexingStore = ({
     const table = schema.tables[tableName];
 
     return db.wrap({ method: `${tableName}.createMany` }, async () => {
-      try {
-        const rows: Row[] = [];
-        await db.transaction().execute(async (tx) => {
-          for (let i = 0, len = data.length; i < len; i += MAX_BATCH_SIZE) {
-            const createRows = data
-              .slice(i, i + MAX_BATCH_SIZE)
-              .map((d) => encodeRow(d, table, kind));
-
-            const _rows = await tx
-              .withSchema(namespaceInfo.userNamespace)
-              .insertInto(tableName)
-              .values(createRows)
-              .returningAll()
-              .execute();
-
-            rows.push(...(_rows as Row[]));
-
-            await tx
-              .withSchema(namespaceInfo.internalNamespace)
-              .insertInto(namespaceInfo.internalTableIds[tableName])
-              .values(
-                createRows.map((row) => ({
-                  operation: 0,
-                  id: row.id,
-                  checkpoint: encodedCheckpoint,
-                })),
-              )
-              .execute();
-          }
-        });
-
-        return rows.map((row) => decodeRow(row, table, kind));
-      } catch (err) {
-        const error = err as Error;
-        throw (kind === "sqlite" &&
-          error.message.includes("UNIQUE constraint failed")) ||
-          (kind === "postgres" &&
-            error.message.includes("violates unique constraint"))
-          ? new StoreError(
-              `Cannot createMany ${tableName} records because one or more records already exist (UNIQUE constraint violation). Hint: Did you forget to await the promise returned by a store method?`,
+      const rows: Row[] = [];
+      await db.transaction().execute(async (tx) => {
+        for (let i = 0, len = data.length; i < len; i += MAX_BATCH_SIZE) {
+          const createRows = data
```

Comment:
Should the transaction be inside of the for loop? Out of scope of the pr but just noticed it